### PR TITLE
chore: update v8 to 0.38.1

### PR DESCRIPTION
**Pull request by lucacasonato**

### CODE REVIEW

**Review by AaronO — APPROVED** [review]

**Comment by lucacasonato** [comment]
@AaronO You approve 700 LOC added deps in cargo.lock?

**Comment by AaronO** [comment]
> @AaronO You approve 700 LOC added deps in cargo.lock?

Was just about to mention that. We shouldn't pull the android related deps into `deno_core`

**Comment by lucacasonato** [comment]
@littledivy Should the added dependencies in rusty_v8 not be in `[target.'cfg(target_os = "android")'.dev-dependencies]` instead of `[target.'cfg(target_os = "android")'.dependencies]`?

**Review by AaronO — CHANGES_REQUESTED** [review]
Lets not bring these android-related deps in, CC @littledivy

File: Cargo.lock [hunk 1]
```diff
@@ -2374,6 +2589,83 @@ dependencies = [
  "thiserror",
 ]
 
+[[package]]
+name = "naga"
```
Comment (bartlomieju) [inline Cargo.lock:2593]: Something is obviously wrong here... we shouldn't need `naga` to build `deno`

**Review by bartlomieju — DISMISSED** [review]
I'm against landing this PR, it seems something went wrong in latest release of `rusty_v8`; there are a bunch of new dependencies that seem completely not necessary.

File: Cargo.lock (hunk 1 above)
Comment (crowlKats) [reply Cargo.lock:2593]: we already do, through webgpu

**Comment by lucacasonato** [comment]
Extra deps are now gone. @AaronO @bartlomieju please take another look.

**Review by AaronO — APPROVED** [review]
LGTM now

**Review by littledivy — APPROVED** [review]
LGTM

**Comment by littledivy** [comment]
Side note: I notice that new rusty_v8 builds don't have the aarch64 android library.

**Comment by lucacasonato** [comment]
@littledivy They are still building.

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**